server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -683,6 +684,14 @@ public boolean isES87TSDBCodecEnabled() {
         Property.Final
     );
 
+    public static final FeatureFlag DOC_VALUES_SPARSE_INDEX = new FeatureFlag("doc_values_sparse_index");
```

Comment:
I moved the feature flag here since we now use it to enable the setting or not.

Reply:
Keep the setting here, but move the setting registration to logsdb plugin?